subsys/modem/modem_cmux.c

```diff
@@ -144,6 +144,62 @@ static void modem_cmux_log_received_frame(const struct modem_cmux_frame *frame)
 	modem_cmux_log_frame(frame, "rcvd", frame->data_len);
 }
 
+#if CONFIG_MODEM_BACKEND_STAT
+static void modem_cmux_init_stat(struct modem_cmux *cmux, const struct modem_cmux_config *config)
+{
+	const struct modem_backend_stat_config stat_config = {
+		.name = "cmux",
+		.transmit_buf_capacity = config->transmit_buf_size,
+		.receive_buf_capacity = config->receive_buf_size,
+	};
+
+	modem_backend_stat_init(&cmux->stat, &stat_config);
+}
+
+static void modem_cmux_advertise_transmit_buf_stat(struct modem_cmux *cmux)
+{
+	const struct modem_backend_stat_adv adv = {
+		.transmit_buf_size = ring_buf_size_get(&cmux->transmit_rb),
+	};
+
+	modem_backend_stat_advertise(&cmux->stat, &adv);
+}
+
+static void modem_cmux_advertise_receive_buf_stat(struct modem_cmux *cmux)
+{
+	const struct modem_backend_stat_adv adv = {
+		.receive_buf_size = cmux->receive_buf_len,
+	};
+
+	modem_backend_stat_advertise(&cmux->stat, &adv);
+}
+
+static void modem_cmux_dlci_init_stat(struct modem_cmux_dlci *dlci,
```

Comment:
Suggest moving this function so that it's not between `modem_cmux_*advertise_*()` functions.